src/miniscript/hash256.rs

```diff
@@ -31,10 +33,18 @@ pub struct Hash([u8; 32]);
 hex_fmt_impl!(Debug, Hash);
 hex_fmt_impl!(Display, Hash);
 hex_fmt_impl!(LowerHex, Hash);
-index_impl!(Hash);
 serde_impl!(Hash, 32);
 borrow_slice_impl!(Hash);
 
+impl<I: SliceIndex<[u8]>> Index<I> for Hash {
+    type Output = I::Output;
+
+    #[inline]
+    fn index(&self, index: I) -> &Self::Output {
+        &self.0[index]
+    }
+}
```

Comment:
I added this because of the trait bounds on `Hash` trait (from `bitcoin_consensus`). `Deref` doesn't help with that, right? Or am I missing something?

Reply:
I don't see any `Hash` trait in `bitcoin_consensus`, I guess you mean `bitcoin_hashes`. You're right that `Deref` doesn't help but it's a question if that bound should be there since it's quite burdensome. I'd have to look into it deeper. The first check is will anything break if we remove it?

Reply:
I'll investigate this, cheers.

Reply:
Please see: https://github.com/rust-bitcoin/bitcoin_hashes/pull/168